EFCoreAssignment/Program.cs

```diff
@@ -14,56 +16,128 @@
 
 var dbContext = new AppDbContext(options);
 
-Filtering(dbContext);
-SingleOrDefault(dbContext);
-LoadingRelatedData_Manual(dbContext);
-LoadingRelatedData_ExplicitLoading(dbContext);
-LoadingRelatedData_EagerLoading(dbContext);
+//Filtering(dbContext);
+//SingleOrDefault(dbContext);
+//LoadingRelatedData_Manual(dbContext);
+//LoadingRelatedData_ExplicitLoading(dbContext);
+//LoadingRelatedData_EagerLoading(dbContext);
+InsertProduct(dbContext);
+InsertProductWithNewShop(dbContext);
+UpdateProduct(dbContext);
+DeleteProduct(dbContext);
+DeleteProductByKey(dbContext);
 
-static void Filtering(AppDbContext dbContext)
+//static void Filtering(AppDbContext dbContext)
+//{
+//// TODO : Filter by Product Name
+//var shop = dbContext.Shops
+//    .Where(s => s.Name == "Nintendo eShop")
+//    .ToList();
+//}
+
+//static void SingleOrDefault(AppDbContext dbContext)
+//{
+//// TODO : Select using SingleOrDefault by Product Id 
+//var prod = dbContext.Products
+//    .SingleOrDefault(p => p.Id == 11);
+//}
+
+//static void LoadingRelatedData_Manual(AppDbContext dbContext)
+//{
+//// TODO : Load Product with related shop data manually
+//var prod = dbContext.Products
+//    .FirstOrDefault();
+//prod.Shop = dbContext.Shops
+//    .Single(s => s.Id == prod.ShopId);
+//}
+
+//static void LoadingRelatedData_ExplicitLoading(AppDbContext dbContext)
+//{
+//// TODO : Load Product with related shop data explicitly
+//var prod = dbContext.Products
+//    .FirstOrDefault();
+//dbContext.Entry(prod)
+//    .Reference(p => p.Shop)
+//    .Load();
+//dbContext.Entry(prod)
+//    .Collection(p => p.Reviews)
+//    .Load();
+//}
+
+//static void LoadingRelatedData_EagerLoading(AppDbContext dbContext)
+//{
+//var output = dbContext.Products
+//    .Include(p => p.Shop)
+//    .Include(p => p.Reviews)
+//    .FirstOrDefault();
+//}
+
+static void InsertProduct(AppDbContext dbContext)
 {
-    // TODO : Filter by Product Name
-    var shop = dbContext.Shops
-        .Where(s => s.Name == "Nintendo eShop")
-        .ToList();
+    // TODO: Insert a new Product
+    var product = new Product()
+    {
+        Name = "Pokemon Scarlet",
+        ShopId = 3
+    };
+    using (dbContext)
+    {
+        dbContext.Add(product);
+        dbContext.SaveChanges();
+    }
 }
 
-static void SingleOrDefault(AppDbContext dbContext)
+static void InsertProductWithNewShop(AppDbContext dbContext)
 {
-    // TODO : Select using SingleOrDefault by Product Id 
-    var prod = dbContext.Products
-        .SingleOrDefault(p => p.Id == 11);
+    // TODO: Insert a new Product with a new Shop
+    var shop = new Shop()
+    {
+        Name = "Bethesda shop"
+    };
+
+    var product = new Product()
+    {
+        Name = "Final Fantasy XIV",
+        ShopId = 8
```

Comment:
Line 93 and 101 might get a conflict since you have created a new shop instance and you also assigned a value to the ShopId. You can remove the ShopId in line 101

Reply:
Replaced `ShopId` to `Shop` and assigned the `shop` variable